source/dubregistry/api.d

```diff
@@ -10,17 +10,30 @@ import dubregistry.registry;
 
 import vibe.d;
 
-DubRegistryWebApi registerDubRegistryWebApi(URLRouter router, DubRegistry registry)
+void registerDubRegistryWebApi(URLRouter router, DubRegistry registry)
 {
-	auto settings = new WebInterfaceSettings;
-	settings.urlPrefix = "/api";
+	auto pkgs = new Packages(registry);
+	router.registerRestInterface(pkgs, "/api/packages");
+}
+
+interface IPackages {
+	@path(":name/latest")
+	string getLatestVersion(string _name);
+
+	@path(":name/stats")
+	Json getStats(string _name);
 
-	auto webapi = new DubRegistryWebApi(registry);
-	router.registerWebInterface(webapi, settings);
-	return webapi;
+	@path(":name/:version/stats")
+	Json getStats(string _name, string _version);
+
+	@path(":name/info")
+	Json getInfo(string _name);
+
+	@path(":name/:version/info")
+	Json getInfo(string _name, string _version);
```

Comment:
There is still the shortcut of using `string getLatest(string id)` instead of using an explicit `@path`. I'm not sure if this is the best long-term approach to modeling collections of obejcts, but it works and is short. I have no strong preference here personally.